File: java/src/main/java/io/fivexer/sdk/model/WorkerStatsResult.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * Per-worker throughput over a window.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerStatsResult {
    private String from;
    private String to;
    private List<WorkerProductivity> workers;

    public String getFrom() { return from; }
    public String getTo() { return to; }
    public List<WorkerProductivity> getWorkers() { return workers; }
}

File: php/src/Model/WorkerStatsResult.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * Per-worker throughput over a window.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerStatsResult
{
    public function __construct(
        public readonly string $from,
        public readonly string $to,
        /** @var list<WorkerProductivity> */
        public readonly array $workers,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            from: (string) ($data['from'] ?? ''),
            to: (string) ($data['to'] ?? ''),
            workers: Json::parseEach($data, 'workers', [WorkerProductivity::class, 'fromArray']),
        );
    }
}
